generator/src/test/java/com/scottlogic/deg/generator/cucumber/interesting/InSet.feature

```diff
@@ -0,0 +1,144 @@
+    Given there is a field foo
+    And foo is anything but null
+    And foo is in set:
+      | "ABC" |
+      | "DEF" |
+      | "GHI" |
+      | "JKL" |
+    Then the following data should be generated:
+      | "foo" |
+      | "ABC" |
+      | "DEF" |
+
+Scenario: Running an 'inSet' request that includes 4 strings with alphabetic characters should return 2 values and null
+    Given there is a field foo
+    And foo is in set:
+      | "ABC" |
+      | "DEF" |
+      | "GHI" |
+      | "JKL" |
+    Then the following data should be generated:
+      | foo   |
+      | null  |
+      | "ABC" |
+      | "DEF" |
+
+Scenario: Running an 'inSet' request that includes only a single string with alphabetic characters should return 1 value and null
+    Given there is a field foo
+    And foo is in set:
+      | "Test" |
+    Then the following data should be generated:
+      | foo    |
+      | null   |
+      | "Test" |
+
+Scenario: Running an 'inSet' request that includes 4 numeric values and not null should return 2 values
+    Given there is a field foo
+    And foo is anything but null
+    And foo is in set:
+      | 10 |
+      | 20 |
+      | 30 |
+      | 40 |
+    Then the following data should be generated:
+      | foo |
+      | 20  |
+      | 40  |
+
+Scenario: Running an 'inSet' request that includes 4 numeric values should return 2 values and null
+    Given there is a field foo
+    And foo is in set:
+      | 100 |
+      | 2000 |
+      | 300 |
+      | 550 |
+    Then the following data should be generated:
+      | foo  |
+      | null |
+      | 100  |
+      | 2000 |
+
+ Scenario: Running an 'inSet' request that includes 4 decimal numeric values should return 2 decimal values and null
+    Given there is a field foo
+    And foo is in set:
+      | 100.05   |
+      | 253.0054 |
+      | 10.8     |
+      | 0.00382  |
+    Then the following data should be generated:
+      | foo      |
+      | null     |
+      | 10.8     |
+      | 253.0054 |
+
+Scenario: Running an 'inSet' request that includes 4 temporal values should return 2 temporal values and null
+    Given there is a field foo
+    And foo is in set:
+      | "2010-01-01T00:00:00.000" |
+      | "2012-01-01T00:01:00.000" |
+      | "2011-01-02T00:00:00.000" |
+      | "2012-01-02T01:10:00.000" |
+    Then the following data should be generated:
+      | foo                       |
+      | null                      |
+      | "2010-01-01T00:00:00.000" |
+      | "2012-01-01T00:01:00.000" |
+
+Scenario: Running an 'inSet' request that has multiple special characters should return 2 values and null
+    Given there is a field foo
+    And foo is in set:
+      | "Cyrillic text: Тхис ис Тест Нумбер 01" |
+      | "Japanese text: これはテスト番号2です"   |
+      | "Korean text: 이것은 시험 번호 3입니다"  |
+    Then the following data should be generated:
+      | foo |
+      | null |
+      | "Japanese text: これはテスト番号2です"   |
+      | "Korean text: 이것은 시험 번호 3입니다"  |
+
```

Comment:
Hi, how does it select which 2 values to return as interesting? In the other examples it seems to be the first 2 values but in this one it is the 2nd and 3rd one.

Reply:
This occurs because a subset is being taken froma HashSet which isn't predictable. These tests are going to be changed.

Reply:
@cakehurstryan raised an good point with which i agree in that, whether the way we select interesting values from the set can be improved to e.g. select boundary values (largest/smalles or shortest/longest or one with unusual characters etc).

Reply:
Should we extend the cucumber syntax to support something like:

```
Then there are _n_ rows emitted
  And there are _n_ rows containing some of:
  | foo |
  | "Japanese text: これはテスト番号2です"   |
  | "Korean text: 이것은 시험 번호 3입니다"  |
```

The _the following data is included in what was generated_ step can be used to detect/assert the presence of null.